crates/roc_std/src/roc_list.rs

```diff
@@ -230,6 +240,44 @@ impl<T> RocList<T> {
                 .cast()
         }
     }
+
+    pub fn append(&mut self, value: T) {
+        self.push(value)
+    }
+
+    pub fn push(&mut self, value: T) {
```

Comment:
Does reserve already deal with slices? (duplicate them to a new allocation even if they are unique). If not, this won't be safe on slices.

Reply:
added the check for seamless slices